rust/template/main.rs

```diff
@@ -49,21 +49,24 @@ fn handle_cmd(hddlog: &HDDlog,
         },
         Command::Commit(record_delta) => {
             // uncomment to enable profiling
-            //PROFILER.lock().unwrap().start("./prof.profile").expect("Couldn't start profiling");
+            // PROFILER.lock().unwrap().start("./prof.profile").expect("Couldn't start profiling");
             let mut current_table = None;
             let res = if record_delta {
-                let cb = if print_deltas {
-                    Some(|table, val: &Record, pol: bool| {
-                        if current_table != Some(table) {
-                            let _ = writeln!(stdout(), "{}:", relid2name(table).unwrap());
-                            current_table = Some(table);
-                        };
-                        let _ = writeln!(stdout(), "{}: {}", *val, if pol { "+1" } else { "-1" });
-                    })
-                } else {
-                    None
-                };
-                hddlog.transaction_commit_dump_changes(cb)
+                hddlog.transaction_commit_dump_changes().map(|changes| if print_deltas {
+                    for (table_id, table_data) in changes.as_ref().iter() {
+                        for (val, weight) in table_data.iter() {
+                            debug_assert!(*weight == 1 || *weight == -1);
+                            if current_table != Some(*table_id) {
```

Comment:
This code can be simplified since you explicitly iterate over tables now (just print table name on each iteration of the outer loop)